### PR TITLE
[Feature] Room DB on Chat rooms screen

### PR DESCRIPTION
**Pull request by luciofm**
- [x] Migrate to androidx/jetpack
- [x] Room DB
- [x] Use LiveData for connection status

Using jetpack, so it depends on Android Studio 3.2-canary17 (you can install it alongside with the current AS)

depends on https://github.com/RocketChat/Rocket.Chat.Kotlin.SDK/pull/173

Currently not working - will be fixed on the next PR:
* Spotlight
* loading and empty screen
* load a channel by ID (for notifications)

### CODE REVIEW

File: app/src/main/java/chat/rocket/android/app/RocketChatApplication.kt [hunk 1]
```diff
@@ -119,118 +115,7 @@ class RocketChatApplication : Application(), HasActivityInjector, HasServiceInje
...
-                    "http://$hostname"
-                } else {
-                    "https://$hostname"
-                }
-            } else {
-                "http://$currentServer"
-            }
-            saveCurrentServerRepository.save(currentServer)
-        }
+        // TODO - remove REALM files.
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/app/RocketChatApplication.kt]: Looks like the REALM files were removed, so you can remove this TODO too and the commented imports

File: app/src/main/java/chat/rocket/android/chatrooms/adapter/RoomMapper.kt [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        } else if (name != null && text != null) {
+            val user = if (localRepository.checkIfMyself(name)) {
+                "${context.getString(R.string.msg_you)}: "
+            } else {
+                "${mapName(name, fullName, unread)}: "
+            }
+
+            val color = if (unread) messageUnreadColor else messageColor
+
+            SpannableStringBuilder()
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/chatrooms/adapter/RoomMapper.kt:102]: Beautiful, huh? ❤️

File: app/src/main/java/chat/rocket/android/chatrooms/di/ChatRoomsFragmentModule.kt [hunk 3]
```diff
@@ -1,23 +1,93 @@
...
     @Provides
+    @PerFragment
     fun provideLifecycleOwner(frag: ChatRoomsFragment): LifecycleOwner {
         return frag
     }
+
+    @Provides
+    @PerFragment
+    @Named("currentServer")
+    fun provideCurrentServer(currentServerInteractor: GetCurrentServerInteractor): String {
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/chatrooms/di/ChatRoomsFragmentModule.kt:43]: 👍 👍

File: app/src/main/java/chat/rocket/android/chatrooms/di/ChatRoomsFragmentModule.kt [hunk 4]
```diff
@@ -1,23 +1,93 @@
...
+        return currentServerInteractor.get()!!
+    }
+
+    @Provides
+    @PerFragment
+    fun provideRocketChatClient(factory: RocketChatClientFactory,
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/chatrooms/di/ChatRoomsFragmentModule.kt:49]: 👍 👍

File: app/src/main/java/chat/rocket/android/chatrooms/infrastructure/ChatRoomsRepository.kt [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    fun getChatRooms(order: Order): LiveData<List<ChatRoom>> {
+        return when(order) {
+            Order.ACTIVITY -> dao.getAll()
+            Order.GROUPED_ACTIVITY -> dao.getAllGrouped()
+            Order.NAME -> dao.getAllAlphabetically()
+            Order.GROUPED_NAME -> dao.getAllAlphabeticallyGrouped()
+        }
+    }
+
+    fun fetchChatRooms() {
```
Comment (philipbrito) [inline app/src/main/java/chat/rocket/android/chatrooms/infrastructure/ChatRoomsRepository.kt]: Empty function.

**Review by philipbrito — COMMENTED** [review]
Thanks for changing the things to the new `androidx`package.

File: app/src/main/java/chat/rocket/android/app/RocketChatApplication.kt (hunk 1 above)
Comment (luciofm) [reply app/src/main/java/chat/rocket/android/app/RocketChatApplication.kt]: Actually this is about the realm database files, I need to investigate the path where they are stored to clean up

**Comment by luciofm** [comment]
@filipedelimabrito synced with `develop`. There are some things that are broken (look the PR description) that will be fixed on the next PR today or tomorrow

**Pull request merged**